providers/okta/resources/policies.go

```diff
@@ -146,57 +145,44 @@ func newMqlOktaPolicy(runtime *resources.Runtime, entry *sdk.PolicyWrapper) (int
 		system = *entry.System
 	}
 
-	settings, err := core.JsonToDict(entry.Settings)
+	settings, err := convert.JsonToDict(entry.Settings)
 	if err != nil {
 		return nil, err
 	}
 
-	return runtime.CreateResource("okta.policy",
-		"id", entry.Id,
-		"name", entry.Name,
-		"description", entry.Description,
-		"priority", entry.Priority,
-		"status", entry.Status,
-		"system", system,
-		"type", entry.Type,
-		"conditions", conditions,
-		"settings", settings,
-		"created", entry.Created,
-		"lastUpdated", entry.LastUpdated,
-	)
+	return CreateResource(runtime, "okta.policy", map[string]*llx.RawData{
+		"id":          llx.StringData(entry.Id),
+		"name":        llx.StringData(entry.Name),
+		"description": llx.StringData(entry.Description),
+		"priority":    llx.IntData(entry.Priority),
+		"status":      llx.StringData(entry.Status),
+		"system":      llx.BoolData(system),
+		"type":        llx.StringData(entry.Type),
+		"conditions":  llx.DictData(conditions),
+		"settings":    llx.DictData(settings),
+		"created":     llx.TimeDataPtr(entry.Created),
+		"lastUpdated": llx.TimeDataPtr(entry.LastUpdated),
+	})
 }
 
 func (o *mqlOktaPolicy) id() (string, error) {
-	id, err := o.Id()
-	if err != nil {
-		return "", err
-	}
-	return "okta.policy/" + id, nil
+	return "okta.policy/" + o.Id.Data, o.Id.Error
 }
 
-func (o mqlOktaPolicy) GetRules() ([]interface{}, error) {
-	op, err := oktaProvider(o.MotorRuntime.Motor.Provider)
-	if err != nil {
-		return nil, err
-	}
-	ctx := context.Background()
-	client := op.Client()
-
-	policyId, err := o.Id()
-	if err != nil {
-		return nil, err
-	}
+func (o mqlOktaPolicy) rules() ([]interface{}, error) {
+	conn := o.MqlRuntime.Connection.(*connection.OktaConnection)
+	client := conn.Client()
 
-	policyType, err := o.Type()
-	if err != nil {
-		return nil, err
+	ctx := context.Background()
+	if o.Id.Error != nil {
+		return nil, o.Id.Error
 	}
 
-	if policyType == ACCESS_POLICY {
-		return getAccessPolicyRules(ctx, o.MotorRuntime, policyId, op.OrganizationID(), op.Token())
+	if o.Type.Data == ACCESS_POLICY {
```

Comment:
Could `o.Type.Error` be not nil?

Above, the code previously checked for `o.Type()` returning an error.

Reply:
Since the field is static, we don't expect errors. 

In fact this is a pattern we will want to ensure down the line, because it leads to a lot of overhead code every provider dev has to do this manually.